src/parser/__init__.py

```diff
@@ -0,0 +1,179 @@
+
+    def parse(self, county, case_number, test): #remove the test value here and just have the tests pass the test data (may have to set path)
+
+        county = county.lower()
+        # get input and output directories and make json dir if not present
+        case_html_path, case_json_path = self.get_directories(county, test)
+        # start 
+        START_TIME_PARSER = time()
+
+        # creating a list of json files already parsed 
+        # TODO: Consider overwriting or deleting all parsed json files. When the parser is in production and parsing the same case ids 
+        # repeatedly on different days, it will need to know that it SHOULD parse the case id again as it will have been updated. 
+        # Perhaps there is a clean up step that removes old parsed json after each run or perhaps it overwrites.
+        cached_case_json_list = [
+            file_name.split(".")[0] for file_name in os.listdir(case_json_path)
+        ]
+
+        # Get a list of the HTML files that it needs to parse. This includes handling of test files.
+        case_html_list = self.get_list_of_html(case_html_path, case_number, county, test)
+
+        #Loops through all of the case html either in the folder where it was scraped, only a single case if specified, or a single test case.
+        for case_html_file_name in case_html_list:
+            try:
+                # This will grab the case id and look for it in the cashed json and won't continue if it exists (unless it's a test)
+                # TODO: Follow up with making updates here depending on if the caching is no longer used. 
+                case_number = case_html_file_name.split(".")[0]
+                if case_number in cached_case_json_list and test == False:
+                    continue
+
+                case_html_file_path = self.get_html_path(case_html_path, case_html_file_name, case_number, test)
+
+                print(f"{case_number} - parsing")
+                with open(case_html_file_path, "r") as file_handle:
+                    case_soup = BeautifulSoup(file_handle, "html.parser", from_encoding="UTF-8")
+
+                # Get the county-specific parser class and method
+                parser_instance, parser_function = self.get_class_and_method(county=county)
+                if parser_instance is not None and parser_function is not None:
+                    case_data = parser_function(county, case_number, case_soup)
+                else:
+                    # Handle the case where parser_instance or parser_function is None
+                    print("Error: Could not obtain parser instance or function.")
+
+                # Adds county field to data
+                case_data['county'] = county
+
+                # Adds a hash to the JSON file of the underlying HTML
+                body = case_soup.find("body")
+                balance_table = body.find_all("table")[-1]
+                if "Balance Due" in balance_table.text:
```

Comment:
Let's add a comment in the code explaining why we do this 

(The "balance due" part of the html changes every day you view the case file, even if nothing else about the case changes. So we don't want that to be part of the unique text that we hash to check if anything has been changed.)

Reply:
Great point! I'll put this in the next update.